src/accelerate/utils/modeling.py

```diff
@@ -397,6 +420,22 @@ def set_module_tensor_to_device(
     else:
         torch.cuda.empty_cache()
 
+    # When handling tied weights, we update tied_params_map to keep track of the tied weights that have already been allocated on the device in
+    # order to avoid duplicating memory, see above.
+    if (
+        tied_params_map is not None
+        and old_value.data_ptr() in tied_params_map
+        and device not in tied_params_map[old_value.data_ptr()]
+    ):
+        tied_params_map[old_value.data_ptr()][device] = new_value
+    elif (
+        value is not None
+        and tied_params_map is not None
+        and value.data_ptr() in tied_params_map
+        and device not in tied_params_map[value.data_ptr()]
+    ):
+        tied_params_map[value.data_ptr()][device] = new_value
```

Comment:
Same reasoning. Note that we need to remove the data `new_value` added here in `tied_params_map` in the `post_forward`, as we are using offloading and want the garbage collector to clear the on-device data.